common/modal.js

```diff
@@ -305,13 +305,20 @@
         vm.versionDate = params.versionDate;
         vm.versionDateRelative = params.versionDateRelative;
         vm.filename = params.displayname;
+        vm.showVersionWarning = params.showVersionWarning;
 
         vm.moreThanWeek = function () {
             var weekAgo = moment().subtract(7, 'days').startOf('day');
             var versionMoment = moment(params.versionDate);
 
             return weekAgo.isAfter(versionMoment);
         }
+
+        if (params.showVersionWarning) {
+            vm.alerts = AlertsService.alerts;
```

Comment:
The current implementation has some issues that by creating a special alert (separated from alert service) we can just avoid.  

One issue is showing the existing alerts in the share modal. if there are some existing alerts (the session expire alert for example), those will be shown in the share modal as well. Another one is the fact that each time users click on the share button, a new alert is being created. So if you keep clicking on “Share” button, they will stack up. Besides I feel like we might prefer only showing this alert in the share modal (The alert description matches the modal and not the main record display).

Instead of reusing the alert service, you can just show an alert box with the same html structure in the `shareCitation.modal.html`. This will duplicate the html structure of alert, but I feel like it will avoid all these unnecessary logic that you have added to alert service (and most probably have to add more to fix those bugs).